infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/MachineResolver.java

```
@@ -96,6 +97,15 @@ private void applyDevfileVolumes(InternalMachineConfig machineConfig) {
     }
   }
 
+  private void applyDevfileEndpoints(InternalMachineConfig machineConfig) {
+    for (org.eclipse.che.api.core.model.workspace.devfile.Endpoint endpoint :
+        component.getEndpoints()) {
+      machineConfig
+          .getServers()
+          .put(endpoint.getName(), ServerConfigImpl.createFromEndpoint(endpoint));
```

Do we have to care about potential clashes here if the plugin already contains an endpoint of that name defined in its `meta.yaml`?

---

hmm, good point. I guess current implementation silently replace that. What do you suggest to do? Fail / prefer one over another ?

---

Override with devfile + warn perhaps.

---

Yes, I incline towards having stronger devfile. Are we already sending some warnings to the workspace loader anywhere? It would be nice to print it in yellow color for example...

---

decided to fail here, when trying to override